Servers/WindowServer/WSCompositor.cpp

```diff
@@ -95,8 +95,10 @@ void WSCompositor::compose()
                 m_wallpaper_mode == WallpaperMode::Unchecked) {
                 m_back_painter->blit(dirty_rect.location(), *m_wallpaper, dirty_rect);
             } else if (m_wallpaper_mode == WallpaperMode::Center) {
-                // TODO: Implement centered wallpaper
-                m_back_painter->blit(dirty_rect.location(), *m_wallpaper, dirty_rect);
+                Point offset{ ws.size().width() / 2 - m_wallpaper->size().width() / 2,
```

Comment:
I think it would be easier to center the wallpaper by doing something like:
```cpp
auto rect = m_wallpaper->rect();
rect.center_within(WSScreen::the().rect());
m_back_painter->blit(rect.location(), *m_wallpaper, m_wallpaper->rect());
```

The compositor will clip things for us, and if clipping isn't working right, we should just fix clipping instead of trying to manually offset things :)

Reply:
Oooo didn't realize that was a thing! I should read the .h files more carefully next time.

Reply:
Ok clipping needs to be fixed. But maybe that's for another PR.